sdk/resource/resource.go

```diff
@@ -51,29 +74,85 @@ func Merge(a, b *Resource) *Resource {
 	if b == nil {
 		return a
 	}
-	res := &Resource{
-		labels: map[core.Key]core.Value{},
+
+	n := len(a.sorted)
+	if len(b.sorted) > len(a.sorted) {
+		n = len(b.sorted)
 	}
-	for k, v := range b.labels {
-		res.labels[k] = v
+	// At a minimum the merge will be as large as the largest resource.
+	s := make([]core.KeyValue, 0, n)
+	k := make(map[core.Key]struct{}, n)
+	ai, bi := 0, 0
+	for ; ai < len(a.sorted) && bi < len(b.sorted); ai, bi = ai+1, bi+1 {
+		akv := a.sorted[ai]
+		k[akv.Key] = struct{}{}
+
+		bkv := b.sorted[bi]
+		if _, ok := k[bkv.Key]; ok {
+			// a overwrites b.
+			s = append(s, akv)
+			continue
+		}
+		k[bkv.Key] = struct{}{}
+
+		// Preserve sort.
+		if kvLess(akv, bkv) {
+			s = append(s, akv)
+			s = append(s, bkv)
+		} else {
+			s = append(s, bkv)
+			s = append(s, akv)
+		}
+	}
```

Comment:
follow order is not addressed.
			a:    kvaaa, kvaaaa, kvaaaaa, kvb
			b:    kv11, kv21, kvb, kvc

Reply:
Ah, yes, good point and good catch. Not sure what I was thinking.

I'm going to just sort after the fact so I don't re-implement the entire `sort` package. Increases the complexity, but I think that is fine given this and the `New` functions are not anticipated to be in a hot path from what I can tell.

Reply:
agreed.

Reply:
Update and added test to catch this. PTAL.